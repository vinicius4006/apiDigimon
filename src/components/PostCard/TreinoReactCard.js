export const TreinoReactCard = ({ name, img, level }) => (
  <div
    className="digiPost shadow-md flex flex-col items-center 
            justify-center transition-all ease-in-out hover:scale-125"
  >
    <img src={img} />
    <div className="digi-content flex flex-col items-center justify-center">
      <h1 className="font-bold">{name}</h1>
      <p>{level}</p>
    </div>
  </div>
);
