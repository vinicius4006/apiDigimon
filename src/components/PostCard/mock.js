export const postCardPropsMock =  
{
    name: 'nameDigimon',
    level: 'levelDigimon',
    img: 'img/digimon.png'
};