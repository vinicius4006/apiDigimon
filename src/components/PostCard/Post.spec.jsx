import { render, screen } from "@testing-library/react";
import { postCardPropsMock } from "./mock";
import { TreinoReactCard } from "./TreinoReactCard";

const props = postCardPropsMock;

describe('<TreinoReactCard />', () => {
    it('should render PostsCard correctly', () => {
      render(<TreinoReactCard {...props}/>)

      expect(screen.getByRole('img', { name: /nameDigimon/i}))
      .toHaveAttribute('src', 'img/digimon.png');
      
     expect(screen.getByRole('heading', { name: /nameDigimon/i }))
     .toBeInTheDocument();

     expect(screen.getByText('levelDigimon')).toBeInTheDocument();
    })

    it('should match snapshot', () => {
        const {container} = render(<TreinoReactCard {...props}/>);
        expect(container.firstChild).toMatchSnapshot();
      
    })
});

//Debug funciona como o console.log()

//Precisamos saber se a imagem está correta e o alt também

//Como não temos um específico para parágrafo, é necessário usar o getByText

//o snapshot é necessário pois ele vai comparar com snap que foi tirado,
//então caso eu adicione algo, pode resultar em erro