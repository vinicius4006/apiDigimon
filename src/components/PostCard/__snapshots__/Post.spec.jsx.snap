// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreinoReactCard /> should match snapshot 1`] = `
<div
  class="digiPost"
>
  <img
    alt="nameDigimon"
    src="img/digimon.png"
  />
  <div
    class="digi-content"
  >
    <h1
      class="h1-content"
    >
      nameDigimon
    </h1>
    <p>
      levelDigimon
    </p>
  </div>
</div>
`;
